### ISSUE TITLE
async afterCallback signature should support terminating request 

### ISSUE THREAD

**Issue by alexmalev** [auth0/nextjs-auth0#952]
### Checklist

- [X] The issue can be reproduced in the [sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [X] I have looked into the [README](https://github.com/auth0/nextjs-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

In version 2.0 the lib now supports terminating requests in afterCallback.

The contract is that afterCallback should not return session and instead do a res.end()

the problem is that the signature of afterCallback should change from:

`export type AfterCallback = (
  req: NextApiRequest,
  res: NextApiResponse,
  session: Session,
  state?: { [key: string]: any }
) => Promise<Session> | Session | undefined;`

to 
`export type AfterCallback = (
  req: NextApiRequest,
  res: NextApiResponse,
  session: Session,
  state?: { [key: string]: any }
) => Promise<Session | undefined> | Session | undefined;`

otherwise we need to //@ts-ignore an async afterCallback that terminates instead of returning a session promise

### Reproduction

```ts
const afterCallback = async (
  req: NextApiRequest,
  res: NextApiResponse,
  session: Session,
  state?: { [key: string]: any }
) => {
  if (session.user.isAdmin) {
    await doSomeAsyncStuff();
    return session;
  }
  res.status(401).end('User is not admin');
};
```

```ts
const handler = handleAuth({
  callback: handleCallback({ afterCallback }),
  })
```

throws: **Type 'Promise<Session | undefined>' is not assignable to type 'Session | Promise<Session> | undefined'.
        Type 'Promise<Session | undefined>' is not assignable to type 'Promise<Session>'**

### SDK version

2.0

### Next.js version

13

### Node.js version

16

**Comment by adamjmcgrath** [comment]
Thanks for raising this @alexmalev - will take a look at fixing this shortly, also happy to accept a PR if you want to expedite it

**Comment by adamjmcgrath** [comment]
Fixed by #954 - will be released in 2.0.1

Thanks @alexmalev!
